core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -245,70 +253,77 @@ export class RteHelperModalComponent {
       }
     }
     this.ngbActiveModal.dismiss(true);
+    this.customizationArgsFormSubscription.unsubscribe();
   }
 
   delete(): void {
     this.ngbActiveModal.dismiss(true);
+    this.customizationArgsFormSubscription.unsubscribe();
   }
 
-  disableSaveButtonForMathRte(): boolean {
-    // This method disables the save button when the Math SVG has not yet
-    // been generated but being processed.
-    if (!this.currentRteIsMathExpressionEditor) {
-      return false;
-    } else {
-      // We know that this is a math rich text component. Hence we can make the
-      // the type more specific.
-      const { value } = this.tmpCustomizationArgs[0] as Extract<
-        CustomizationArgsNameAndValueArray[number],
-        { name: 'math_content' }
-      >;
-      return value.mathExpressionSvgIsBeingProcessed || value.raw_latex === '';
-    }
-  }
-
-  disableSaveButtonForLinkRte(): boolean {
-    // This method disables the save button when the `text` field for the
-    // Link RTE looks like a URL but it does not match the `url`. Otherwise,
-    // creators can make the `url` a malicious website and make the `text`
-    // a safe website.
-    if (!this.currentRteIsLinkEditor) {
-      return false;
-    }
-    // We know that this is a link rich text component. Hence we can make the
-    // the type more specific.
-    const tmpCustomizationArgs = this.tmpCustomizationArgs as Extract<
-      CustomizationArgsNameAndValueArray[number],
-      { name: 'url' | 'text' }
-    >[];
-    let url: string = tmpCustomizationArgs[0].value;
-    let text: string = tmpCustomizationArgs[1].value;
-
-    // First check if the `text` looks like a URL.
-    const suffixes = ['.com', '.org', '.edu', '.gov'];
-    let textLooksLikeUrl = false;
-    for (const suffix of suffixes) {
-      if (text.endsWith(suffix)) {
-        textLooksLikeUrl = true;
+  onCustomizationArgsFormChange(value: number | string | boolean): void {
+    if (this.componentId === this.COMPONENT_ID_MATH) {
+      let rawLatex: string = value[0].raw_latex;
+      let mathExpressionSvgIsBeingProcessed: boolean = (
+        value[0].mathExpressionSvgIsBeingProcessed);
+      this.saveButtonIsDisabled = (
+        mathExpressionSvgIsBeingProcessed || rawLatex === '');
+    } else if (this.componentId === this.COMPONENT_ID_VIDEO) {
+      let start: number = value[1];
+      let end: number = value[2];
+      if (start === 0 && end === 0) {
+        this.saveButtonIsDisabled = false;
+      } else {
+        this.saveButtonIsDisabled = (start >= end);
       }
-    }
-    if (!textLooksLikeUrl) {
-      return false;
-    }
-    // If the text looks like a URL, strip the leading 'http://' or
-    // 'https://' or 'www.'.
-    const prefixes = ['https://', 'http://', 'www.'];
-    for (const prefix of prefixes) {
-      if (url.startsWith(prefix)) {
-        url = url.substring(prefix.length);
+      if (this.saveButtonIsDisabled) {
+        this.updateRteErrorMessage(
```

Comment:
If we ensure that the error message shows if and only if the save button is disabled, can we get rid of this.saveButtonIsDisabled altogether and use a function like isErrorMessageNonempty() to detemine whether the button is disabled? That way there is a single source of truth for this, and you can just set the error message as soon as an issue is detected (which conceptually makes more sense then setting saveButtonIsDisabled first and then the error message later, since the save button can be disabled for various reasons).

Reply:
Implemented a solution for this, however I am not 100% sure I understood it correctly about the implementation since it toggles the state of the button in the following way
```ts
  isErrorMessageNonEmpty(): boolean {
    return this.errorMessage !== '';
  }
```
which made me use this awkward approach for math rte
```ts
      if (mathExpressionSvgIsBeingProcessed || rawLatex === '') {
        this.updateRteErrorMessage(' ');
      } else {
        this.updateRteErrorMessage('');
      }
```
maybe this is fine or you have any easy changes to make it work in the appropriate way?